snapcraft/project/_project_options.py

```diff
@@ -197,49 +200,35 @@ def deb_arch(self):
     def kernel_arch(self):
         return self.__machine_info["kernel"]
 
-    @property
-    def local_plugins_dir(self):
-        deprecated_plugins_dir = os.path.join(self.parts_dir, "plugins")
-        if os.path.exists(deprecated_plugins_dir):
-            handle_deprecation_notice("dn2")
-            return deprecated_plugins_dir
-        return os.path.join(self.__project_dir, "snap", "plugins")
-
-    @property
-    def parts_dir(self):
-        return os.path.join(self.__project_dir, "parts")
-
-    @property
-    def stage_dir(self):
-        return os.path.join(self.__project_dir, "stage")
-
-    @property
-    def prime_dir(self):
-        return os.path.join(self.__project_dir, "prime")
-
-    @property
-    def debug(self):
-        return self.__debug
-
     def __init__(
         self,
         use_geoip=False,
         parallel_builds=True,
         target_deb_arch=None,
         debug=False,
         *,
-        project_dir: str = None
+        work_dir: str = None
     ) -> None:
 
-        if project_dir is None:
-            self.__project_dir = os.getcwd()
-        else:
-            self.__project_dir = project_dir
+        # Here for backwards compatibility.
+        project_dir = os.getcwd()
+        if work_dir is None:
+            work_dir = project_dir
+
+        self.use_geoip = use_geoip
+        self.parallel_builds = parallel_builds
+        self.debug = debug
+
+        self.parts_dir = os.path.join(work_dir, "parts")
+        self.stage_dir = os.path.join(work_dir, "stage")
+        self.prime_dir = os.path.join(work_dir, "prime")
+        self.local_plugins_dir = _get_local_plugins_dir(project_dir, self.parts_dir)
```

Comment:
What happens if a plugin decides to change these?

Reply:
A plugin can change anything in our code base, if they change it breaks, it is the deal with using python.

Reply:
They couldn't change any of these before this PR without a hack, and we're now making them public attributes, which makes them not only available for writing but implicitly says it's okay to do so. Making breakage possible is one thing (we can't avoid that with Python), but inviting it with the API design is another. APIs should be designed to be easy to use correctly, and difficult to use incorrectly.

Reply:
We already had this argument with ProjectInfo, why bring it back again with such passion? :-)

Reply:
This is public API. Moving to attributes opens it up as widely as possible-- we can't take that back. I disagree with it, and it's my duty to say so, but you can of course override with no further argument from me.

Reply:
issue taken care of